src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,343 @@
+  this.$overlay = null
+  this.keypressCounter = 0
+  this.lastKeyWasModified = false
+  this.timeoutTime = 5000 // milliseconds
+
+  // Store the timeout events so that we can clear them to avoid user keypresses overlapping
+  // setTimeout returns an id that we can use to clear it with clearTimeout,
+  // hence the 'Id' suffix
+  this.keypressTimeoutId = null
+  this.timeoutMessageId = null
+}
+
+/**
+ * Create the <span> we use for screen reader announcements.
+ */
+ExitThisPage.prototype.initUpdateSpan = function () {
+  this.$updateSpan = document.createElement('span')
+  this.$updateSpan.setAttribute('role', 'status')
+  this.$updateSpan.className = 'govuk-visually-hidden'
+
+  this.$module.appendChild(this.$updateSpan)
+}
+
+/**
+ * Create button click handlers.
+ */
+ExitThisPage.prototype.initButtonClickHandler = function () {
+  // Main EtP button
+  this.$button.addEventListener('click', this.exitPage.bind(this))
+
+  // EtP skiplink
+  if (this.$skiplinkButton) {
+    this.$skiplinkButton.addEventListener('click', this.exitPage.bind(this))
+  }
+}
+
+/**
+ * Create the HTML for the 'three lights' indicator on the button.
+ */
+ExitThisPage.prototype.buildIndicator = function () {
+  // Build container
+  // Putting `aria-hidden` on it as it won't contain any readable information
+  this.$indicatorContainer = document.createElement('div')
+  this.$indicatorContainer.className = 'govuk-exit-this-page__indicator'
+  this.$indicatorContainer.setAttribute('aria-hidden', 'true')
+
+  // Create three 'lights' and place them within the container
+  for (var i = 0; i < 3; i++) {
+    var $indicator = document.createElement('div')
+    $indicator.className = 'govuk-exit-this-page__indicator-light'
+    this.$indicatorContainer.appendChild($indicator)
+  }
+
+  // Append it all to the module
+  this.$button.appendChild(this.$indicatorContainer)
+}
+
+/**
+ * Update whether the lights are visible and which ones are lit up depending on
+ * the value of `keypressCounter`.
+ */
+ExitThisPage.prototype.updateIndicator = function () {
+  // Show or hide the indicator container depending on keypressCounter value
+  if (this.keypressCounter > 0) {
+    this.$indicatorContainer.classList.add('govuk-exit-this-page__indicator--visible')
+  } else {
+    this.$indicatorContainer.classList.remove('govuk-exit-this-page__indicator--visible')
+  }
+
+  // Turn out all the lights
+  var $lightsOn = this.$indicatorContainer.querySelectorAll('.govuk-exit-this-page__indicator-light--on')
+  nodeListForEach($lightsOn, function ($light) {
+    $light.classList.remove('govuk-exit-this-page__indicator-light--on')
+  })
+
+  // Turn on the ones we want on
+  var $lightsQueried = this.$indicatorContainer.querySelectorAll('.govuk-exit-this-page__indicator-light')
+  nodeListForEach($lightsQueried, function ($light, index) {
+    if (index < this.keypressCounter) {
+      $light.classList.add('govuk-exit-this-page__indicator-light--on')
+    }
+  }.bind(this))
```

Comment:
Suggestion: This could potentially be done in a single pass by using [`DOMTokenList.toggle`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle):

```suggestion
  // Turn on only the indicators we want on
  var $indicators = this.$indicatorContainer.querySelectorAll(
    '.govuk-exit-this-page__indicator-light'
  )

  nodeListForEach($indicators, function ($indicator, index) {
    $indicator.classList.toggle(
      'govuk-exit-this-page__indicator-light--on',
      index < this.keypressCounter
    )
  }.bind(this))
```

Reply:
[Toggle has fairly meagre browser support](https://caniuse.com/?search=toggle). Are we ok putting in a polyfill?

Reply:
Doh! My fault for not checking my work 🤦🏻 I'll do this then!